File: direct_code/project_7/app/src/main/java/com/example/project_7/MainActivity.java
package com.example.project_7;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;

import android.Manifest;
import android.app.Activity;
import android.content.ContentProviderOperation;
import android.content.OperationApplicationException;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.os.RemoteException;
import android.provider.ContactsContract;
import android.view.View;
import android.widget.EditText;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private EditText editTextName,editTextNumber;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    //THESE LINE OF CODE WILL GIVE THE PERMISSION TO WRITE INTO THE CONTACT
        ActivityCompat.requestPermissions(this,
                new String[]{Manifest.permission.WRITE_CONTACTS},
                PackageManager.PERMISSION_GRANTED);

        editTextName = findViewById(R.id.editTextTextPersonName);
        editTextNumber = findViewById(R.id.editTextNumber);
    }

    public  void buttonAddContact(View view){
        ArrayList<ContentProviderOperation> contentProviderOperations
                = new ArrayList<ContentProviderOperation>();

        contentProviderOperations.add(ContentProviderOperation.newInsert(
                        ContactsContract.RawContacts.CONTENT_URI)
                .withValue(ContactsContract.RawContacts.ACCOUNT_TYPE, null)
                .withValue(ContactsContract.RawContacts.ACCOUNT_NAME, null)
                .build());

        // Adding Name
        contentProviderOperations.add(ContentProviderOperation
                .newInsert(ContactsContract.Data.CONTENT_URI)
                .withValueBackReference(ContactsContract.Data.RAW_CONTACT_ID,0)
                .withValue(ContactsContract.Data.MIMETYPE,
                        ContactsContract.CommonDataKinds.StructuredName.CONTENT_ITEM_TYPE)
                .withValue(ContactsContract.CommonDataKinds.StructuredName.DISPLAY_NAME,
                        editTextName.getText().toString())
                .build());

        // Adding Number
        contentProviderOperations.add(ContentProviderOperation
                .newInsert(ContactsContract.Data.CONTENT_URI)
                .withValueBackReference(ContactsContract.Data.RAW_CONTACT_ID,0)
                .withValue(ContactsContract.Data.MIMETYPE,
                        ContactsContract.CommonDataKinds.Phone.CONTENT_ITEM_TYPE)
                .withValue(ContactsContract.CommonDataKinds.Phone.NUMBER,
                        editTextNumber.getText().toString())
                .withValue(ContactsContract.CommonDataKinds.Phone.TYPE,
                        ContactsContract.CommonDataKinds.Phone.TYPE_MOBILE)
                .build());

        try {
            getContentResolver().applyBatch(ContactsContract.AUTHORITY, contentProviderOperations);
        } catch (OperationApplicationException e) {
            e.printStackTrace();
        } catch (RemoteException e) {
            e.printStackTrace();
        }

    }
}

File: direct_code/lab_record_1/app/src/main/java/com/example/lab_record_1/MainActivity.kt
package com.example.lab_record_1

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var name = findViewById<EditText>(R.id.name)
        val reg = findViewById<EditText>(R.id.reg)
        val submit = findViewById<Button>(R.id.submit)


        submit.setOnClickListener {
            var enteredName = name.text.toString()
            var enteredreg = reg.text.toString()
            if (enteredName=="")
            {
                //grettingTextView.text = ""
                Toast.makeText(
                    this@MainActivity,
                    "please enter your name",
                    Toast.LENGTH_SHORT
                ).show()
            }
            else if ((enteredreg=="" )|| (enteredreg.length < 10))
            {
                //grettingTextView.text = ""
                Toast.makeText(
                    this@MainActivity,
                    "please enter your registration no of 10 character",
                    Toast.LENGTH_SHORT
                ).show()
            }
            else{
                Toast.makeText(
                    this@MainActivity,
                    "Thank you " + enteredName + " registered successfully",
                    Toast.LENGTH_SHORT

                ).show()
                val intent = Intent(this,MainActivity2::class.java)
                intent.putExtra("USER",enteredName)
                startActivity(intent)
            }

        }
    }
}